packages/jellyfish-api-core/__tests__/category/loan/getInterest.test.ts

```diff
@@ -0,0 +1,191 @@
+      minColRatio: 200,
+      interestRate: new BigNumber(3),
+      id: 'scheme'
+    })
+    await testing.generate(1)
+
+    const vaultAddress = await testing.generateAddress()
+    const vaultId = await testing.rpc.loan.createVault({
+      ownerAddress: vaultAddress,
+      loanSchemeId: 'scheme'
+    })
+    await testing.generate(1)
+
+    await testing.rpc.loan.depositToVault({
+      vaultId: vaultId, from: collateralAddress, amount: '10000@DFI'
+    })
+    await testing.generate(1)
+
+    await testing.rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '1000@TSLA'
+    })
+    await testing.generate(1)
+
+    interestTSLABlockHeight = await testing.rpc.blockchain.getBlockCount()
+
+    await testing.rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '50@UBER'
+    })
+    await testing.generate(1)
+
+    await testing.rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: '250@AMZN'
+    })
+    await testing.generate(1)
+  }
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+    await setup()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should getInterest', async () => {
+    const interests = await testing.rpc.loan.getInterest('scheme')
+    /**
+     * output:
+     *
+     * [{
+     *  token: 'TSLA',
+     *  totalInterest: 0.00000285,
+     *  interestPerBlock: 0.00000057
+     * }]
+     */
+    expect(interests.length).toStrictEqual(3)
+    expect(interests[0].token).toStrictEqual('TSLA')
+    expect(interests[1].token).toStrictEqual('UBER')
+    expect(interests[2].token).toStrictEqual('AMZN')
+
+    // calculate interest per block
+    const netInterest = (3 + 0) / 100 // (scheme.rate + loanToken.interest) / 100
+    const blocksPerDay = (60 * 60 * 24) / (10 * 60) // 144 in regtest
+    const interestPerBlock = (netInterest * 1) / (365 * blocksPerDay) //  netInterest * rate.count / 365 * blocksPerDay
+    expect(interests[0].interestPerBlock.toFixed(8)).toStrictEqual(interestPerBlock.toFixed(8))
+
+    // calculate total interest
+    const blockHeight = await testing.rpc.blockchain.getBlockCount()
+    const totalInterest = ((blockHeight - interestTSLABlockHeight + 1) * interestPerBlock)
+    expect(interests[0].totalInterest.toFixed(8)).toStrictEqual(totalInterest.toFixed(8))
+  })
+
+  it('should getInterest with token', async () => {
+    // name
+    {
+      const interests = await testing.rpc.loan.getInterest('scheme', 'TSLA')
+      expect(interests.length).toStrictEqual(1)
+      expect(interests[0].token).toStrictEqual('TSLA')
+    }
+
+    // creationTx
+    {
+      const interests = await testing.rpc.loan.getInterest('scheme', txUber)
+      expect(interests.length).toStrictEqual(1)
+      expect(interests[0].token).toStrictEqual('UBER')
+    }
+
+    // id
+    {
+      const interests = await testing.rpc.loan.getInterest('scheme', '3')
```

Comment:
would be better if we can call RPC to get the id

Reply:
actually this style is more on readability wise
the point is not needed to know `3` is `AMZN` token but rather to tell this rpc support tokenId filter